src/app/modules/item/pages/group-progress-grid/group-progress-grid.component.html

```diff
@@ -51,13 +51,11 @@
           {{ row.header }}
         </td>
         <td *ngFor="let col of columns; let colIndex = index">
-          <ng-container *ngIf="row.data[colIndex]">
-            <alg-user-progress
-              [userProgress]="row.data[colIndex]"
-              [canAccess]="data.can_access"
-              (permEditionRequested)="onAccessPermissions(row.header, row.id, col.id)"
-            ></alg-user-progress>
-          </ng-container>
+          <alg-user-progress
+            [userProgress]="row.data[colIndex]"
+            [canAccess]="data.can_access"
+            (permEditionRequested)="onAccessPermissions(row.header, row.id, col.id)"
```

Comment:
If you remove this check, it means you display much more "user-progress" (with undefined data) instead of leaving the cell blank. I don't see how it is an improvement.

Reply:
When removing this ngIf, it takes one second less in the worst case scenario, that's because the `alg-user-progress` displays a blank cell on its own and does the checks in the component, not the template. Apparently JS `if` statements are way faster than `ngIf`s, the gain is here.
After re-checking the mapping, what's more problematic is that `row.data[colIndex]` can be undefined but `userProgress` cannot and is asserted defined. I'll revert this change.
Fixed.